test/test_nn.py

```diff
@@ -1051,36 +1051,50 @@ def compare_cpu_gpu(outputs_cpu, outputs_gpu):
             for bias in (True, False):
                 for bidirectional in (False, True):
                     for batch_first in (False, True):
-                        num_directions = 2 if bidirectional else 1
-                        if batch_first:
-                            input_val = torch.randn(batch, seq_length, input_size)
-                        else:
-                            input_val = torch.randn(seq_length, batch, input_size)
-                        hx_val = torch.randn(num_layers * num_directions, batch, hidden_size)
-
-                        rnn = module(input_size,
-                                     hidden_size,
-                                     num_layers,
-                                     bias=bias,
-                                     dropout=dropout,
-                                     bidirectional=bidirectional,
-                                     batch_first=batch_first)
-
-                        outputs_cpu = forward_backward(
-                            False, rnn, input_val, hx_val, rnn.all_weights)
-
-                        rnn_gpu = module(input_size,
+                        for skip_input in (False, True):
+                            num_directions = 2 if bidirectional else 1
+                            if skip_input:
+                                input_val = torch.randn(seq_length, batch, hidden_size)
+                            else:
+                                input_val = torch.randn(seq_length, batch, input_size)
+                            if batch_first:
+                                input_val = input_val.transpose(0, 1).contiguous()
+                            hx_val = torch.randn(num_layers * num_directions, batch, hidden_size)
+
+                            rnn = module(hidden_size if skip_input else input_size,
                                          hidden_size,
                                          num_layers,
                                          bias=bias,
                                          dropout=dropout,
                                          bidirectional=bidirectional,
-                                         batch_first=batch_first)
-
-                        outputs_gpu = forward_backward(
-                            True, rnn_gpu, input_val, hx_val, rnn.all_weights)
-
-                        compare_cpu_gpu(outputs_cpu, outputs_gpu)
+                                         batch_first=batch_first,
+                                         skip_input=skip_input)
+
+                            # cuDNN bug, bias still used even when skip_input true
+                            if skip_input and bias:
```

Comment:
From my memory this is just a problem for skip_input as @ngimel highlighted, but I may be wrong...